result/src/commonTest/kotlin/com/github/kittinunf/result/ResultTest.kt

```diff
@@ -269,6 +269,27 @@ class ResultTest {
         assertFalse(hasFailureChanged)
     }
 
+    @Test
+    fun `should be able to transform both value from 2 sides`() {
+        val s = Result.of<String, Throwable> {
+            "hello world"
+        }
+
+        val transformedSuccess = s.mapBoth({ it.length }, { IllegalStateException("changed!") })
+
+        assertIs<String>(s.getOrNull())
+        assertEquals(11, transformedSuccess.get())
+        assertEquals(null, transformedSuccess.getFailureOrNull())
+
+        val f = Result.of<Int, Throwable> { 1 / 0 } //this should have arithmetic exception
+
+        val transformedFailure = f.mapBoth({ it > 1}, { RuntimeException("changed!") })
+
+        assertIs<ArithmeticException>(f.getFailureOrNull())
+        assertEquals(null, transformedFailure.getOrNull())
+        assertEquals("changed!", transformedFailure.getFailureOrNull()!!.message)
```

Comment:
What about to assert the `transformedFailure`'s type as well?

Reply:
True, I will fix this 👍

Reply:
Fixed! 5650e4e